### PR TITLE
[MERGE][MOV] mail, various: split and move mail code by model

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
[MOV] various

Move code, views and data of mail / rating models into their own files.
Purpose is to have one file / model, easier when searching for inherited parts
of the code and see at a glance various usage of those models.

[FIX] in ir_cron

Trigger methods are defined as api.model but actually use self. Those are
therefore not model. Smart.

LINKS

Spotted during Task-2207626 (Rating: Delay rating notification to ease feedback)

### CODE REVIEW

File: odoo/addons/base/models/ir_cron.py [hunk 1]
```diff
@@ -431,7 +431,6 @@ def toggle(self, model, domain):
         active = bool(self.env[model].search_count(domain))
         return self.try_write({'active': active})
 
-    @api.model
```
Comment (Julien00859) [inline odoo/addons/base/models/ir_cron.py:434]: roooh le noob :see_no_evil:

File: odoo/addons/base/models/ir_cron.py [hunk 2]
```diff
@@ -459,7 +458,6 @@ def _trigger(self, at=None):
 
         self._trigger_list(at_list)
 
-    @api.model
```
Comment (Julien00859) [inline odoo/addons/base/models/ir_cron.py:462]: :see_no_evil:

**Review by Julien00859 — APPROVED** [review]

File: odoo/addons/base/models/ir_cron.py (hunk 1 above)
Comment (tde-banana-odoo) [reply odoo/addons/base/models/ir_cron.py:434]: :D

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Pull request closed without merging**